src/EntityComponentManager.cc

```diff
@@ -74,10 +74,37 @@ class ignition::gazebo::EntityComponentManagerPrivate
   /// \brief Flag that indicates if all entities should be removed.
   public: bool removeAllEntities{false};
 
+  /// \brief True if the entityComponents map was changed.  Primarily used
+  /// by the multithreading functionality in `State()` to allocate work to
+  /// each thread.
+  public: bool entityComponentsDirty{true};
+
+  /// \brief The number of threads to run entity component computations.
+  /// Initialized to 1, but calculated later based on the hardware thread
+  /// support and number of components each thread is allowed to handle, see
+  /// `componentsPerThread`
+  public: uint64_t numComponentThreads = 1;
```

Comment:
All of the new thread logic is only used by the `State` function. I think this could be made clearer in variable / function names and also the documentation. The `State` function is an optional part of the ECM, only used by GUI, distributed sim and logging. So I think it's worth isolating this logic a bit more so that it's clearer it applies only to `State`. In the future, we may want to parallelize other aspects of the ECM and add other threads.

Reply:
I updated docs 2e253210ed78695db9da45ffb8e025d9985c6d75 but couldn't think of more specific names, I feel like those names explain what exactly they do, do you think we should be more specific with them belonging to `State` function?